core/tauri/src/manager.rs

```diff
@@ -303,6 +303,17 @@ impl<R: Runtime> WindowManager<R> {
       registered_scheme_protocols.push("tauri".into());
     }
     if !registered_scheme_protocols.contains(&"asset".into()) {
+      let window_url = Url::parse(&pending.url).unwrap();
+      let window_origin = format!(
+        "{}://{}{}",
+        window_url.scheme(),
+        window_url.host().unwrap(),
+        if let Some(port) = window_url.port() {
+          format!(":{}", port)
+        } else {
+          "".into()
+        }
+      );
```

Comment:
This works as expected running `yarn tauri build`, but doesn't work when running `yarn tauri dev`. Same CORS error is displayed on the screen. Don't know how to fix, but my guess is that it is not registering http://localhost:3000 from the vite dev command. I have changed the allowed origin line in my fork to "*" and it works correctly without CORS error. 

https://github.com/antoniotorres/tauri/blob/fix/asset-protocol-cors/core/tauri/src/manager.rs#L307

Reply:
ooops i didn't realize i was reassign the url after defining the window origin